docs/src/content/ignition/docs/guides/viem.md

```diff
@@ -0,0 +1,149 @@
+import "@nomicfoundation/hardhat-ignition-viem";
+```
+
+:::
+
+:::tab{value="JavaScript"}
+
+```javascript
+require("@nomicfoundation/hardhat-ignition-viem");
+```
+
+:::
+
+::::
+
+:::tip
+
+Only **one** Hardhat Ignition package should be imported within your Hardhat config file. Viem and ethers support cannot both be enabled at the same time.
+
+:::
+
+## The Ignition object
+
+The `@nomicfoundation/hardhat-plugin-viem` plugin adds an `ignition` object to the [Hardhat Runtime Environment](../../../hardhat-runner/docs/advanced/hardhat-runtime-environment.md).
+
+The `ignition` object exposes a `deploy` method, which takes an Ignition module, deploys it against the current network (in tests this will be the in-memory Hardhat network), and returns the results of the module as typed Viem contract instances.
+
+The `deploy` method takes the Module as its first argument:
+
+```typescript
+const ApolloModule = buildModule("Apollo", (m) => {
+  const apollo = m.contract("Rocket", ["Saturn V"]);
+
+  return { apollo };
+});
+
+it("should have named the rocket Saturn V", async function () {
+  const { apollo } = await hre.ignition.deploy(ApolloModule);
+
+  assert.equal(await apollo.read.name(), "Saturn V");
+});
+```
+
+The `ignition.deploy` method returns an object with a Viem contract instance per `Future` returned as a result in the module.
+
+## Usage with Artifacts
+
+To infer the type of a Viem contract instance, the full type of the ABI must be available when initializing a `Future` within a module.
+
+For named Hardhat contracts, the ABI and type information will be retrieved automatically using [hardhat-viem's type generation](https://hardhat.org/hardhat-runner/docs/advanced/using-viem#contract-type-generation).
+
+If you are passing an artifact object to initialize a `Future` within a module, you will need to ensure the artifact's ABI is either declared with [const assertions](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-4.html#const-assertions) or defined inline:
+
+```typescript
+const counterArtifact = {
+  // ...
+  abi: [...] as const, // <--- const assertion
+}
+
+const CounterModule = buildModule("Counter", (m) => {
+  const counter = m.contract("Counter", CounterArtifact, [42]);
+
+  return { counter };
+});
+```
+
+If type inference isn't working for the returned contract, it is likely that the ABI type needs a `const` assertion or to be defined inline.
+
+This is because the ABI type must be as narrow as possible for Viem to infer the correct type for the contract, and by default Typescript will broaden the types within object literals.
+
+:::tip
+
+Typescript doesn't support importing JSON as const, it will instead automatically broaden the type, breaking Viem's inference. The artifact must be defined inline within Typescript.
+
+:::
+
+See [Viem's type inference requirements](https://viem.sh/docs/typescript.html#type-inference) for more details.
+
+## Sending transactions with a different account
```

Comment:
Let's add a :::tip in the general testing guide, and create an issue to revisit the testing docs so that it is clearer that there are two "flavors"

Reply:
To clarify, sending transactions with a different account is the same in both the ethers and viem variants (the input parameters are the same across both versions, it is only the result that varies). I was suggesting moving the "Sending transactions with a different account" to the main testing section. Which on reflection, I am more convinced is the right move.

Can I check that I am following your comment correctly:

1. that we open an issue saying we should restructure the guides so that the current testing guide (assuming ethers) + viem stuck on as an extra, becomes, ethers testing guide + viem testing guide
2. But for the moment, add a tooltip to the general testing guide indicating there is a Viem option

Reply:
> that we open an issue saying we should restructure the guides so that the current testing guide (assuming ethers) + viem stuck on as an extra, becomes, ethers testing guide + viem testing guide

Yes. Or maybe have two entirely different testing guides. We have to think about it.

> But for the moment, add a tooltip to the general testing guide indicating there is a Viem option

correct.

And given that situation, I don't think the duplication of the "sending transactions with a different account" is too bad.

Reply:
I have added the tooltip. I also duplicated the "Sending transactions with a different account" in the general test guide and made it work with ethers.